drizzlepac/haputils/svm_quality_analysis.py

```diff
@@ -516,6 +516,304 @@ def compare_ra_dec_crossmatches(hap_obj, json_timestamp=None, json_time_since_ep
+        default value is logical 'None'
+
+    log_level : int, optional
+        The desired level of verboseness in the log statements displayed on the screen and written to the
+        .log file. Default value is 'NOTSET'.
+
+    Returns
+    -------
+    Nothing
+    """
+    # Initiate logging!
+    log.setLevel(log_level)
+
+    # Check to make sure there's at last 2 filter-level products. If not, return.
+    num_filter_prods = 0
+    for total_obj in total_obj_list:
+        num_filter_prods += len(total_obj.fdp_list)
+    if num_filter_prods == 1:
+        log.error("At least 2 filter-level products are necessary to perform analysis of inter-filter "
+                  "cross matched sources. Only 1 filter-level product was found. ")
+        return
+    elif num_filter_prods == 0:
+        log.error("At least 2 filter-level products are necessary to perform analysis of inter-filter "
+                  "cross matched sources. No filter-level products were found. ")
+        return
+    else:
+        log.info("Found {} filter-level products for use in analysis of inter-filter cross matched source"
+                 " positions:".format(num_filter_prods))
+        ctr = 1
+        for total_obj in total_obj_list:
+            for filt_obj in total_obj.fdp_list:
+                log.info("{}: {}".format(ctr, filt_obj.drizzle_filename))
+                ctr += 1
+
+        filtobj_dict = {}
+        xmatch_ref_imgname = None
+        max_sources = 0
+
+        for total_obj in total_obj_list:
+            for filt_obj in total_obj.fdp_list:
+                log.info("{} {} {}".format(">"*20, filt_obj.drizzle_filename, "<"*20))
+                filtobj_dict[filt_obj.drizzle_filename] = find_hap_point_sources(filt_obj, log_level=log_level)
+                n_sources = len(filtobj_dict[filt_obj.drizzle_filename]['sources'])
+                log.info("Identified {} sources in {}".format(n_sources, filt_obj.drizzle_filename))
+                if n_sources > max_sources:
+                    max_sources = n_sources
+                    xmatch_ref_imgname = filt_obj.drizzle_filename
+        log.info("Crossmatch reference image {} contains {} sources.".format(xmatch_ref_imgname, max_sources))
+        xmatch_ref_catname = xmatch_ref_imgname[:-8] + "point-cat-fxm.ecsv"
+
+        # Perform coord transform and write temp cat files for cross match
+        temp_cat_file_list = []
+        for imgname in filtobj_dict.keys():
+            filtobj_dict[imgname] = transform_coords(filtobj_dict[imgname],
+                                                     xmatch_ref_imgname,
+                                                     log_level=log_level)
+            temp_cat_name = imgname[:-8] + "point-cat-fxm.ecsv"
+            temp_cat_file_list.append(temp_cat_name)
+            filtobj_dict[imgname]["sources"].write(temp_cat_name, format="ascii.ecsv")
```

Comment:
It would simpler (and less error-prone?) to add an entry for `cat_file` to `filtobj_dict[imgname]` to store the value of `temp_cat_name` since it will be needed later in the processing (see Line 615).  This would make sure that the name only gets constructed in one place for use everywhere else, so that the rest of the code does not need to know how or what was used to make that name, it can just simply use the stored value.

Reply:
Issue resolved in commit [b7fb0b3](https://github.com/spacetelescope/drizzlepac/pull/719/commits/b7fb0b33a8f7ad925bcb1451d9021de938d055c5)